vaadin-spring/src/main/java/com/vaadin/flow/spring/VaadinSpringSecurity.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2000-2021 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.spring;
+
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.builders.WebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+import org.springframework.security.web.util.matcher.OrRequestMatcher;
+import org.springframework.security.web.util.matcher.RequestMatcher;
+
+/**
+ * Helpers for Spring Security configuration of Vaadin applications.
+ *
+ * Applications that use Spring Security typically need to bypass security
+ * checks for static resources. This class contains helper methods for setting
+ * up a {@link WebSecurityConfigurerAdapter} that allows static resources and
+ * the app shell to retrieved without authentication while requiring
+ * authentication for routes.
+ *
+ * @see EnableWebSecurity
+ * @see WebSecurityConfigurerAdapter
+ * @see WebSecurityConfigurerAdapter#configure(WebSecurity)
+ *
+ */
+public class VaadinSpringSecurity {
+
+    private VaadinSpringSecurity() {
+    }
+
+    /**
+     * Matcher for app shell (index page) and framework internal requests.
+     *
+     * @return default {@link HttpSecurity} bypass matcher
+     */
+    public static RequestMatcher getDefaultHttpSecurityPermitMatcher() {
+        return new OrRequestMatcher(Stream.of(
+                "/",
+                "/vaadinServlet/**")
+                .map(AntPathRequestMatcher::new)
+                .collect(Collectors.toList()));
+    }
+
+    /**
+     * Matcher for Vaadin static resources.
+     *
+     * @return default {@link WebSecurity} ignore matcher
+     */
+    public static RequestMatcher getDefaultWebSecurityIgnoreMatcher() {
+        return new OrRequestMatcher(Stream.of(
+                "/VAADIN/**",
```

Comment:
/VAADIN/ ist dangerous, see https://github.com/vaadin/spring/issues/678#issuecomment-736337987 and the comments above.